extensions/amp-story/1.0/utils.js

```diff
@@ -273,8 +273,7 @@ export function setHistoryState(win, stateName, value) {
     ...state,
     [stateName]: value,
   };
-
-  history.replaceState(newHistory, '');
+  throttle(win, () => history.replaceState(newHistory, ''), 1000)();
```

Comment:
This has one bug + one thing that will come back to bite us in the future:

1. it is creating a new "throttler" everything single time, and executing it right away. If `setHistoryState` is called again right away, a new "throttler" will be created and triggered, when it should have been blocked
2. Even if we throttle the calls to `replaceState`, the most recent `historyState` must be returned by `getHistoryState`. Otherwise, we can introduce race conditions like, for example:
- Get history state -> add page to navigation path -> setHistoryState
- Get history state -> add page to navigation path -> setHistoryState
- Get history state (retrieved the one from step 1, as step 2 was throttled) -> add page to navigation path -> setHistoryState (missing the page added in step 2)

Reply:
My bad, 1 should be solved now.

For 2, that's true but I think this is not going to happen in practice since getHistoryState only happens at initialization, meaning a user would have to navigate very quickly then refresh the page quickly to get at this point...

Reply:
1. We should throttle all calls to `replaceState`
2. You're right, but it is very easy to solve, and 10 minutes now could save a day of debugging/getting context/fixing down the line :)